### PR TITLE
Add carlberg function

### PR DESCRIPTION
**Pull request by manuhuth**
Add Carlberg function to optimization.

### CODE REVIEW

**Comment by manuhuth** [comment]
The next step is to replace the function from the lecture notes on optimization with the one from temfpy.

File: temfpy/optimization.py [hunk 1]
```diff
@@ -179,3 +179,77 @@ def rosenbrock(x):
...
+    :math:`b>0` the function is no longer convex and has many local
+    minima (see first graph below), making it hard for
+    local optimization methods to find the global minimum,
+    which is still at
+    :math:`x = \begin{pmatrix}1 & \dots & 1 \end{pmatrix}^T`.
+
+    .. figure:: ../../docs/_static/images/fig-carlberg_noise.png
+       :align: center
+    .. figure:: ../../docs/_static/images/fig-carlberg_no_noise.png
+       :align: center
```
Comment (amageh) [inline temfpy/optimization.py:223]: Maybe you can add a title for both plots? You can infer which one is which from the notes, but titles might still be even clearer.

File: docs/_static/codes/fig-carlberg.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        plotted at the first dimension.
+    x2 : array_like
+        Input vector with points at which the function should be
+        plotted at the second dimension.
+    a : array_like
+        Input vector with dimension :math:`2` passed to carlberg.
+    b : integer
+        Scaling factor of the sinusoidal terms passed to carlberg.
+    save_name : string
+                Name under which the graph will be saved,
```
Comment (amageh) [inline docs/_static/codes/fig-carlberg.py:31]:
```suggestion
                Name under which the graph will be saved.

```

**Review by amageh — APPROVED** [review]
Very nice!

**Pull request merged**